test_dev.sh

```diff
@@ -0,0 +1,169 @@
+}
+
+function print_usage_exit() {
+  echo "$USAGE" 1>&2
+  exit 0
+}
+
+function print_invalid_option_missing_argument_exit() {
+  echo "Invalid Option: $OPTARG requires an argument, use: -h for help" 1>&2
+  exit 1
+}
+
+function print_invalid_option_exit() {
+  echo "Invalid Option: $OPTARG use: -h for help" 1>&2
+  exit 1
+}
+
+function print_list_of_integration_tests() {
+  for file in ./test/integration/*.go; do
+    [ -e "$file" ] || continue
+    cat "$file" | grep -e '^func Test' | awk '{print $2}' | sed s/\(t//
+  done
+  exit 0
+}
+
+USAGE="$(cat -- <<-EOM
+Usage:
+    -h             Displays this help message
+    -a             Run all lint, unit, and integration tests
+
+Commands:
+    unit           Unit tests subcommand, run unit -h for more information
+    integration    Integration tests subcommand, run integration -h for more information
+EOM
+)"
+
+while getopts ":ha" opt; do # Parse options to the `test_cmd` command
+  case "${opt}" in
+    h) 
+      print_usage_exit
+      ;;
+    a)
+      trap_outcome_on_exit
+      docker-compose run --use-aliases boulder ./test.sh
+      ;;
+    *)
+      print_invalid_option_exit
+      ;;
+  esac
+done
+
+if [ $# -gt 1 ] # Check for subcommands
+then
+  shift "$((OPTIND -1))"
+  subcommand="${1}" && shift
+
+USAGE="$(cat -- <<-EOM
+Usage:"
+    -h                Displays this help message
+    -a                Run all unit tests
+    -d <DIRECTORY>    Run unit tests for a specific directory
+EOM
+)"
+
+  case "$subcommand" in
+  unit) # Parse options to the unit sub command
+    while getopts ":d:ha" opt; do
+      case "${opt}" in
+        h) 
+          print_usage_exit
+          ;;
+        a)
+          trap_outcome_on_exit
+          docker-compose run --use-aliases boulder go test -p 1 ./...
+          ;;
+        d)
+          trap_outcome_on_exit
+          docker-compose run --use-aliases boulder go test "$OPTARG"
+          ;;
+        :) # <DIRECTORY>
+          print_invalid_option_missing_argument_exit
+          ;;
+        *)
+          print_invalid_option_exit
+          ;;
+      esac
+    done
+    shift "$((OPTIND -1))"
+    ;;
+esac
+
+USAGE="$(cat -- <<-EOM
+Usage:"
+    -h                   Displays this help message"
+    -a                   Run all integration tests"
+    -l                   List of available integration tests"
+    -f <FILTER_REGEX>    Run only those tests and examples matching the regular expression"
+
+                         Note:"
+                         This option disables the '"back in time"' integration test setup"
+
+                         For tests, the regular expression is split by unbracketed slash (/)"
+                         characters into a sequence of regular expressions"
+
+                         Example:"
+                         ./"$(basename -- "$0")" integration -f TestAkamaiPurgerDrainQueueFails/TestWFECORS"
+EOM
+)"
+
+case "$subcommand" in
+  integration) # Parse options to the integration sub command
```

Comment:
Just realized that this wrapper doesn't currently have a flag to easily switch the integration tests back and forth between running against `config` vs running against `config-next`. That sort of high-level "which version of the tests are you running" is probably worth pulling into a top-level integration subcommand flag, alongside -a and -l.